src/native/linux_wayland.rs

```diff
@@ -140,18 +140,108 @@ unsafe extern "C" fn seat_handle_capabilities(
 ) {
 }
 
+#[cfg(not(all(any(target_arch = "x86", target_arch = "x86_64"), target_os = "linux")))]
```

Comment:
is it really a necessity? Would be nice to keep wayland support on non-x86 targets

Reply:
Support should still be there. cfg allows you to define two functions in the code but only compile one, depending on the platform.. That is what I did here, for both of the problematic functions.

Reply:
@not-fl3 You should be able to use u8 on all platforms, that is what the [docs](https://doc.rust-lang.org/std/ffi/struct.CStr.html) say, but the error (shown below) says otherwise. Perhaps this is a problem with the Rust std library?

![image](https://user-images.githubusercontent.com/101361189/174901596-6e3593e1-66fb-40ca-8cfd-86c2b5fc2788.png)

Reply:
yeah, I do not really understand why its i8/u8

but it kinda makes sense: 
```
It can be constructed safely from a &[[u8](https://doc.rust-lang.org/std/primitive.u8.html)] slice, or unsafely from a raw *const c_char
```

we are using unsafe `from_ptr` that accepts `c_char`

Reply:
and from the practical point of view the only important part of the doc is `from_ptr` signature: 

```rust
pub unsafe fn from_ptr<'a>(ptr: *const c_char) -> &'a CStr
```

its a `c_char`. `c_char` is either i8 or u8: https://doc.rust-lang.org/std/os/raw/type.c_char.html

so we should just ensure that its `c_char` in our export "C" functions

Reply:
Could you update the PR with the `c_char` (basically with the diff a few comments above)? I'll merge it right away :)